### PR TITLE
update-index now checks for updates for local packs

### PR DESCRIPTION
**Pull request by bgn42**

### CODE REVIEW

File: cmd/installer/root.go [hunk 1]
```diff
@@ -420,6 +420,50 @@ func UpdateInstalledPDSCFiles(pidxXML *xml.PidxXML, concurrency int, timeout int
 		}
 	}
 
+	pdscFiles, err = utils.ListDir(Installation.LocalDir, ".pdsc$")
```
Comment (jkrech) [inline cmd/installer/root.go:423]: I am not familiar with ListDir, however why is there a `$` behind .pdsc?

File: cmd/installer/root.go [hunk 2]
```diff
@@ -420,6 +420,50 @@ func UpdateInstalledPDSCFiles(pidxXML *xml.PidxXML, concurrency int, timeout int
...
+		return err
+	}
+
+	numPdsc = len(pdscFiles)
+	if utils.GetEncodedProgress() {
+		log.Infof("[J%d:F\"%s\"]", numPdsc, Installation.PublicIndex)
+	}
+
+	for _, pdscFile := range pdscFiles {
+		log.Debugf("Checking if \"%s\" needs updating", pdscFile)
```
Comment (jkrech) [inline cmd/installer/root.go:434]: More precisely you are checking whether you can find and read a pdsc file to update the copy in .Local 
Needs updating sounds as if you can actually check whether there is a newer version available.

File: cmd/installer/root.go [hunk 3]
```diff
@@ -1184,6 +1228,51 @@ func (p *PacksInstallationType) downloadPdscFile(pdscTag xml.PdscTag, skipInstal
...
+func (p *PacksInstallationType) loadPdscFile(pdscTag xml.PdscTag, timeout int) error {
+	basePdscFile := fmt.Sprintf("%s.%s.pdsc", pdscTag.Vendor, pdscTag.Name)
+	pdscFilePath := filepath.Join(p.LocalDir, basePdscFile)
+
+	pdscURL := pdscTag.URL
+
+	log.Debugf("Loading %s from \"%s\"", basePdscFile, pdscURL)
+
+	pdscFileURL, err := url.Parse(pdscURL)
+	if err != nil {
```
Comment (jkrech) [inline cmd/installer/root.go:1240]: We agreed that we should simply ignore an empty url. 
Are we flagging info warning or error if a url is valid but the file constructed pdsc file path does not exist?
I think warning is sufficient.

**Review by jkrech — APPROVED** [review]
See comments. Please answer my questions.

File: cmd/installer/root.go [hunk 4]
```diff
@@ -420,6 +420,50 @@ func UpdateInstalledPDSCFiles(pidxXML *xml.PidxXML, concurrency int, timeout int
...
 	}
 
+	pdscFiles, err = utils.ListDir(Installation.LocalDir, ".pdsc$")
+	if err != nil {
+		return err
+	}
+
+	numPdsc = len(pdscFiles)
+	if utils.GetEncodedProgress() {
+		log.Infof("[J%d:F\"%s\"]", numPdsc, Installation.PublicIndex)
```
[the remark this answers is not in the document]
Comment (bgn42) [reply cmd/installer/root.go:430]: Yes, is wrong. Copy-paste ...

File: cmd/installer/root.go (hunk 1 above)
Comment (bgn42) [reply cmd/installer/root.go:423]: This is a regular expression. $ means that there should be nothing behind ".pdsc"

File: cmd/installer/root.go (hunk 2 above)
Comment (bgn42) [reply cmd/installer/root.go:434]: I check at the end, if the copied pdsc file contains another highest version than the old one in .local. Nearly the same as in line 371 with the public pdsc files.

File: cmd/installer/root.go (hunk 3 above)
Comment (bgn42) [reply cmd/installer/root.go:1240]: Upps, overseen. I now skip them in the new line 443

**Review by jkrech — APPROVED** [review]
LGTM.

**Pull request merged**